### PR TITLE
Scala linter

### PR DESCRIPTION
**Pull request by rr83019**
Add [scalafix](https://scalacenter.github.io/scalafix/) linter
Add [scalafmt](https://scalameta.org/scalafmt/) formatter
Add [sbt-java-formatter](https://github.com/sbt/sbt-java-formatter) formatter

### CODE REVIEW

File: .scalafmt.conf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+version=2.5.2 
+maxColumn = 80
```
Comment (hrj) [inline .scalafmt.conf:2]: 80 is too conservative. I think 120 would be better. And if that becomes too much, we can tighten it later.

File: src/main/scala/lc/captchas/LabelCaptcha.scala [hunk 2]
```diff
@@ -16,33 +16,34 @@ class LabelCaptcha extends ChallengeProvider {
...
-    val knownImageFile = knownFiles(r)
-    val unknownImageFile = unknownFiles(s)
-    val ip = new ImagePair(knownImageFile, unknownImageFile)
+  def returnChallenge(): Challenge =
+    synchronized {
+      val r = scala.util.Random.nextInt(knownFiles.length)
+      val s = scala.util.Random.nextInt(unknownFiles.length)
+      val knownImageFile = knownFiles(r)
+      val unknownImageFile = unknownFiles(s)
+      new ImagePair(knownImageFile, unknownImageFile)
```
Comment (hrj) [inline src/main/scala/lc/captchas/LabelCaptcha.scala:32]: If this `ImagePair` is not being used, it can be removed.

**Comment by hrj** [comment]
Excellent! This was a much needed change.

I think setting `maxColumns` to 120 will reduce the number of line splits. So I will wait for that before merging.

**Comment by rr83019** [comment]
> I think setting `maxColumns` to 120 will reduce the number of line splits. So I will wait for that before merging.

I have updated the `maxColumns` length, but it hasn't had a significant impact on the code formatting. Maybe because it breaks lines exceeding the limit and doesn't format lines within the limit.

**Comment by hrj** [comment]
> > I think setting `maxColumns` to 120 will reduce the number of line splits. So I will wait for that before merging.
> 
> I have updated the `maxColumns` length, but it hasn't had a significant impact on the code formatting. Maybe because it breaks lines exceeding the limit and doesn't format lines within the limit.

Ok, then can you change this PR to only have the config changes? When that is merged, you can raise a separate PR with code changes based on the config. That way, only few code lines will change.

Also, do these style rules get checked automatically when a PR is raised?  If not, you can make that check a part of this PR, or a separate PR if its complicated.

**Pull request closed without merging**